app/models/service_template_transformation_plan_request.rb

```diff
@@ -0,0 +1,25 @@
+class ServiceTemplateTransformationPlanRequest < ServiceTemplateProvisionRequest
+  TASK_DESCRIPTION = 'VM Transformations'.freeze
+
+  delegate :transformation_mapping, :vm_requests, :to => :source
+
+  def requested_task_idx
+    vm_requests.where(:status => 'Approved')
+  end
+
+  def customize_request_task_attributes(req_task_attrs, vm_request)
+    req_task_attrs[:source] = vm_request.resource
+  end
+
+  def source_vms
+    vm_requests.where(:status => %w(Queued Failed)).pluck(:resource_id)
+  end
+
+  def validate_vm(_vm_id)
+    true
+  end
+
+  def approve_vm(vm_id)
```

Comment:
Who is the caller of this?  Is it the UI (through API?)

Reply:
automate calls it. The actual logic is to be enhanced. I will add a comment.